lib/vyasa/medium/event.ex

```diff
@@ -0,0 +1,40 @@
+defmodule Vyasa.Medium.Event do
+  use Ecto.Schema
+
+  import Ecto.Changeset
+
+  alias Vyasa.Written.{Verse, Source}
+  alias Vyasa.Medium.Voice
+
+  @primary_key {:id, Ecto.UUID, autogenerate: true}
+  schema "events" do
+    field :origin, :integer
+    field :duration, :integer
+    field :phase, :string
+
+    embeds_many :fragments, EventFrag do
+      field :offset, :integer
+      field :duration, :integer
+      field :quote, :string
+      field :status, :string
+    end
+
+    belongs_to :verse, Verse, foreign_key: :verse_id, type: :binary_id
+    belongs_to :voice, Voice, foreign_key: :voice_id, type: :binary_id
+    belongs_to :source, Source, foreign_key: :source_id, type: :binary_id
+  end
+
+  @doc false
+  def changeset(event, attrs) do
+    event
+    |> cast(attrs, [:origin, :duration, :phase, :verse_id, :voice_id, :source_id])
+    |> cast_embed(:fragments, with: &frag_changeset/2)
+    |> validate_required([:origin, :duration, :fragments, :source_id])
```

Comment:
note: not going to be touching fragments for the next bit 

we don't have sub-verse breakdown

@rtshkmr replicate the timestamp and css thing, investigate the issue

Reply:
todo: mull on the protocol for fragment creation, for a more fine-tuned UI experience for verse-highlight

options: 
1. consider musical stuff (e.g. beat) and use a gradient for this 
2. do based on character count or something


++ phases, consider adding "refrain" ==> for instrumentals and no verses (may not be the correct term) 
this allows verse id to empty and highlighting and stuff doesn't prematurely highlight the verse 
this will also allow fragments to be faked 



++ investigat e state of the art for these fragment creations (e.g. in karaoke websites)

Reply:
To investigate when doing the next immediate PR